tests/test.spec.ts

```diff
@@ -0,0 +1,294 @@
+            step.inputs[1],
+            step.inputs[2]
+          );
+        } else {
+          await modifyFile(
+            page,
+            step.inputs[0],
+            step.inputs[1],
+            step.inputs[2],
+            step.inputs[3]
+          );
+        }
+        break;
+      case 'getByLocator-save':
+        const locatorVal = await page.locator(step.inputs[0]).allInnerTexts();
+        saved.push(locatorVal);
+        break;
+      case 'clickByRole':
+        await page.getByRole(step.inputs[0], { name: step.inputs[1] }).click();
+        break;
+      case 'walletApproveConnect':
+        await walletConnect(context);
+        break;
+      case 'walletApprove':
+        await walletApprove(context);
+        break;
+      case 'checkIfIsIncremented':
+        console.log('SAVED:', saved);
+        console.log('INITAL INPUT:', parseInt(step.inputs[0]));
+        console.log('FINAL INPUT:', parseInt(step.inputs[1]));
+        checkIfIsIncremented(
+          parseInt(step.inputs[0]),
+          parseInt(step.inputs[1])
+        );
+        break;
+      default:
+    }
+  }
+}
+
+async function clickCopyButton(page: Page, id: string) {
+  let clipboardText = { text: '', output: '' };
+  const selector = `#${id} + div > div > div > button[aria-label="Copy to Clipboard"]`;
+  await page.locator(selector).click();
+  const rawText: string = await page.evaluate('navigator.clipboard.readText()');
+  if (rawText.startsWith('$')) {
+    clipboardText = separateCommand(rawText);
+  } else {
+    clipboardText.text = rawText;
+    clipboardText.output = '';
+  }
+  return clipboardText;
+}
+
+async function runCommand(page: Page, buttonName: string, goToFolder?: string) {
+  const copied = await clickCopyButton(page, buttonName);
+  console.log('COPIED', copied.text);
+  let command = copied.text;
+  if (goToFolder) {
+    if (goToFolder.includes('<COMMAND>')) {
+      command = goToFolder.replace('<COMMAND>', copied.text);
+    } else {
+      command = goToFolder + copied.text;
+    }
+  }
+  console.log('COMMAND', command);
+  const commandOutput = execSync(command, {
+    encoding: 'utf-8',
+  });
+  console.log('COMMAND OUTPUT', commandOutput);
+  if (copied.output !== '') {
+    compareOutputs(commandOutput, copied.output);
+  }
+}
+
+async function writeToFile(page: Page, buttonName: string, filePath: string) {
+  const content = await clickCopyButton(page, buttonName);
+  fs.writeFileSync(filePath, content.text + '\n\n');
+}
+
+async function modifyFile(
+  page: Page,
+  buttonName: string,
+  filePath: string,
+  atLine?: number,
+  removeLines?: number[]
+) {
+  const content = await clickCopyButton(page, buttonName);
+  if (!atLine && !removeLines) {
+    fs.appendFileSync(filePath, content.text + '\n\n');
+  } else {
+    const lines = fs.readFileSync(filePath, 'utf8').split('\n');
+    if (removeLines) {
+      removeLines.forEach((lineNumber) => {
+        lines[lineNumber - 1] = '~~~REMOVE~~~';
```

Comment:
Why we need this markup `'~~~REMOVE~~~'`?

Reply:
It just makes it easier to think about the `atLine` prop (which inserts code at a certain line) so you don't need to factor in the lines being removed.